src/duke/Ui.java

```diff
@@ -0,0 +1,301 @@
+package duke;
+import java.io.File;
+import java.lang.reflect.Array;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Ui {
+    boolean exited = false;
+    TaskList tl;
+    Storage s;
```

Comment:
Would it be better if you use more meaningful names instead of abbreviation?

Reply:
Yes single letter variables should only be used in loops.